### PR TITLE
Cpplint additional

### PR DESCRIPTION
**Pull request by phlptp**
### Summary

If merged this pull request will add some more of the cpplint tests and fix the resulting issues

### CODE REVIEW

**Comment by phlptp** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones removed
==============
+ src/helics/network/tcp/TcpCommsSS.cpp  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/GMLC-TDC/HELICS/pullRequest?prid=5343571&bid=17494781)

**Comment by phlptp** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones removed
==============
+ src/helics/network/tcp/TcpCommsSS.cpp  -1
+ src/helics/network/tcp/TcpComms.cpp  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/GMLC-TDC/HELICS/pullRequest?prid=5343571&bid=17494781)

**Comment by phlptp** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones removed
==============
+ src/helics/network/tcp/TcpComms.cpp  -1
+ src/helics/network/tcp/TcpCommsSS.cpp  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/GMLC-TDC/HELICS/pullRequest?prid=5343571&bid=17494781)

**Comment by phlptp** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones added
============
- tests/helics/shared_library/FilterTests.cpp  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/GMLC-TDC/HELICS/pullRequest?prid=5343571&bid=17494781)

**Comment by phlptp** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones added
============
- tests/helics/shared_library/FilterTests.cpp  8
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/GMLC-TDC/HELICS/pullRequest?prid=5343571&bid=17494781)

File: src/helics/core/coreTypeOperations.cpp [hunk 1]
```diff
@@ -162,45 +164,45 @@ namespace core {
...
-#    define ZMQ_AVAILABILITY true
+    static bool constexpr zmq_availability{true};
 #endif
 
 #ifndef ENABLE_MPI_CORE
-#    define MPI_AVAILABILITY false
+    static bool constexpr mpi_availability{false};
 #else
-#    define MPI_AVAILABILITY true
+    static bool constexpr mpi_availability{false};
```
Comment (nightlark) [inline src/helics/core/coreTypeOperations.cpp:175]:
```suggestion
    static bool constexpr mpi_availability{true};
```

Comment (phlptp) [reply src/helics/core/coreTypeOperations.cpp:175]: clang-tidy recognizing that the shared_ptr was not necessary in that context and did nothing except an extra increment.

**Comment by phlptp** [comment]
I stared  at the shared-ptr thing for a while, but I think it is correct that it doesn't do much since the intent was to ensure the lifetime of the object, which is done by the outer call and if the internal function doesn't need it there is no reason to copy the shared_ptr.

File: tests/helics/shared_library/FilterTests.cpp [hunk 2]
```diff
@@ -36,16 +36,16 @@ TEST_P(filter_type_tests, registration)
     auto mFed = GetFederateAt(1);
 
     helicsFederateRegisterGlobalEndpoint(mFed, "port1", "", &err);
-    helicsFederateRegisterGlobalEndpoint(mFed, "port2", NULL, &err);
+    helicsFederateRegisterGlobalEndpoint(mFed, "port2", nullptr, &err);
     EXPECT_EQ(err.error_code, helics_ok);
     CE(auto f1 = helicsFederateRegisterFilter(fFed, helics_filter_type_custom, "filter1", &err));
     CE(helicsFilterAddSourceTarget(f1, "port1", &err));
-    EXPECT_TRUE(f1 != NULL);
+    EXPECT_TRUE(f1 != nullptr);
```
Comment (nightlark) [inline tests/helics/shared_library/FilterTests.cpp:43]: Interesting that it didn't suggest `EXPECT_NE` for these.

Comment (phlptp) [reply tests/helics/shared_library/FilterTests.cpp:43]: I suspect it doesn't trigger off pointers, possibly only built-in types like ints,  it also doesn't suggest it for things like fed_id which is a simple wrapper around an integer.

**Pull request merged**